lib/ssh_scan/target_parser.rb

```diff
@@ -1,33 +1,50 @@
 require 'netaddr'
 require 'string_ext'
+require 'set'
 
 module SSHScan
   class TargetParser
-    def enumerateIPRange(ip, port = "22")
+    def enumerateIPRange(ip, port = "22", port_append = false)
       if ip.fqdn?
-        socket = ip.concat(":").concat(port.to_s)
-        return [socket]
+        socket = port_append ? ip.concat(":").concat(port.to_s) : ip
+        return [socket].to_set
       else
         if ip.include? "-"
           octets = ip.split('.')
           range = octets.pop.split('-')
           lower = NetAddr::CIDR.create(octets.join('.') + "." + range[0])
           upper = NetAddr::CIDR.create(octets.join('.') + "." + range[1])
           ip_array = NetAddr.range(lower, upper,:Inclusive => true)
-          ip_array.map! { |ip| ip.concat(":").concat(port.to_s) }
-          return ip_array
+          ip_array.map! { |ip| port_append ? ip.concat(":").concat(port.to_s) : ip }
+          return ip_array.to_set
         elsif ip.include? "/"
           cidr = NetAddr::CIDR.create(ip)
           ip_array = cidr.enumerate
           ip_array.delete(cidr.network)
           ip_array.delete(cidr.last)
-          ip_array.map! { |ip| ip.concat(":").concat(port.to_s) }
-          return ip_array
+          ip_array.map! { |ip| port_append ? ip.concat(":").concat(port.to_s) : ip }
+          return ip_array.to_set
         else
-          socket = ip.concat(":").concat(port.to_s)
-          return [socket]
+          socket = port_append ? ip.concat(":").concat(port.to_s) : ip
+          return [socket].to_set
         end
       end
     end
+
+    # Remove instances of ip, where ip:port already exist.
+    # This would mean that port was somehow passed. Otherwise,
+    # make ip = ip:22
+    def sanitizeIPRange(ip_range)
```

Comment:
@jinankjain yes

Maybe I got you confused. Let me clarify once more.

Right now, when you pass github.com -p 99, it checks github.com:22, github.com:99, which is a side effect of appending ports in -p to the targets given in -t. Since the default port is 22, the first time you were doing enumerateIPRange, it would convert github.com to github.com:22, and then simply append 99 to the host github.com again. But then this 22 is kind of redundant.

The need for my logic stems from the fact that there is (maybe) no way to check whether -p was passed within -t block in OptionParser's code. Thus, enumerate in my PR converts github.com -p 99 to github.com, github.com:99, and then sanitizes it to remove github.com since a github.com:port argument is present. Maybe there are shorter ways to do this, and please feel free to indicate them, I'd be more than happy to change my logic to something shorter.

Sorry for the long explanation :)